### PR TITLE
Add custom parameters into Server class

### PR DESCRIPTION
**Pull request by dzenciarz**
Hi David,
I added options and retries parameters into Server class which are directly mapped into imap_open function - in some cases it's very useful to use this parameters.

Best regards,
Jakub Hudak

### CODE REVIEW

File: src/Server.php [hunk 1]
```diff
@@ -38,12 +48,16 @@
...
+     * @param int $retries       Retries number
      * @param array  $parameters Connection parameters
      */
     public function __construct(
         string $hostname,
         string $port = '993',
         string $flags = '/imap/ssl/validate-cert',
+        int $options = 0,
+        int $retries = 1,
         array $parameters = []
```
Comment (Slamdunk) [inline src/Server.php]: Please move the new options at the end of the signature to maintain backward compatibility

File: src/Server.php [hunk 2]
```diff
@@ -79,8 +95,8 @@ public function authenticate(string $username, string $password): ConnectionInte
             $this->getServerString(),
             $username,
             $password,
-            0,
-            1,
+            $this->options,
+            $this->retries,
```
Comment (Slamdunk) [inline src/Server.php]: We need tests: maybe we can try the `CL_EXPUNGE` flag to prove this functionality

File: src/Server.php (hunk 1 above)
Comment (dzenciarz) [reply src/Server.php]: Done.

File: src/Server.php (hunk 2 above)
Comment (dzenciarz) [reply src/Server.php]: I'll try to prepare tests during weekend to prove this changes actually works

File: src/Server.php [hunk 3]
```diff
@@ -31,20 +31,34 @@
...
     /**
      * Constructor.
      *
      * @param string $hostname   Internet domain name or bracketed IP address
      *                           of server
      * @param string $port       TCP port number
      * @param string $flags      Optional flags
+     * @param int $options       Connection options
+     * @param int $retries       Retries number
      * @param array  $parameters Connection parameters
```
Comment (Slamdunk) [inline src/Server.php]: Parameters should be ordered accordingly to method signature.
Moreover you need to run `vendor/bin/php-cs-fixer fix -v`

**Pull request closed without merging**